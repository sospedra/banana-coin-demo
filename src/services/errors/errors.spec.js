/* global expect, describe, it */
const errorsService = require('./')

jest.mock('pretty-error', () => {
  return class PrettyError {
    skipNodeFiles () {}
    skipPackage () {}
    render () { jest.fn() }
  }
})

const createRes = function () {
  this.send = jest.fn((payload) => payload)
  this.status = jest.fn(() => this)
  return this
}

describe('service errors', () => {
  it('should return a collection of errors responses', () => {
    const errors = errorsService.sendError(createRes())

    expect(errors).toMatchSnapshot()
    Object.values(errors).forEach((send) => {
      expect(send()).toMatchSnapshot()
    })
  })

  it('should create a middleware with logs and a critical response', () => {
    const res = createRes()
    errorsService.errorMiddleware(new Error(), null, res)

    expect(res.status).toHaveBeenCalledWith(500)
  })
})
