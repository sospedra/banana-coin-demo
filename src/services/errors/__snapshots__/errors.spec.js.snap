// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`service errors should return a collection of errors responses 1`] = `
Object {
  "badCredentials": [Function],
  "critical": [Function],
  "missing": [Function],
  "unauth": [Function],
  "unsecure": [Function],
  "version": [Function],
}
`;

exports[`service errors should return a collection of errors responses 2`] = `
Object {
  "error": "Bad auth credentials",
}
`;

exports[`service errors should return a collection of errors responses 3`] = `
Object {
  "error": "Unknown internal critical error",
}
`;

exports[`service errors should return a collection of errors responses 4`] = `
Object {
  "error": "Route not found. And never been here",
}
`;

exports[`service errors should return a collection of errors responses 5`] = `
Object {
  "error": "Missing or invalid Authorization header",
}
`;

exports[`service errors should return a collection of errors responses 6`] = `
Object {
  "error": "Missing or invalid API token",
}
`;

exports[`service errors should return a collection of errors responses 7`] = `
Object {
  "error": "Unsupported version number",
}
`;
